lib/srfi/128/comparators.scm

```diff
@@ -1,12 +1,26 @@
 
 (define-record-type Comparator
-  (make-comparator type-test equality ordering hash)
+  (%make-comparator% type-test equality ordering hash)
   comparator?
   (type-test comparator-type-test-predicate)
   (equality comparator-equality-predicate)
   (ordering comparator-ordering-predicate)
   (hash comparator-hash-function))
 
+(define (make-comparator type-test equality ordering hash)
+  (%make-comparator%
+    type-test
+    equality
+    ordering
+    (if (or (opcode? hash)
+            (not (procedure? hash))
+            (procedure-variadic? hash)
+            (not (= (procedure-arity hash) 1)))
```

Comment:
How about just `(> (procedure-arity hash) 1)`?

Reply:
I figured this made more sense because the specific case it was fixing was a 1-argument function, but it doesn't really matter (0-argument functions will fail no matter what). Done.